tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/validation/TclMissingFixtureValidatorTest.xtend

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.tcl.dsl.validation
+
+import org.eclipse.xtext.common.types.JvmOperation
+import org.eclipse.xtext.common.types.JvmParameterizedTypeReference
+import org.eclipse.xtext.common.types.JvmType
+import org.eclipse.xtext.validation.ValidationMessageAcceptor
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentCaptor
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.testeditor.aml.AmlFactory
+import org.testeditor.aml.InteractionType
+import org.testeditor.aml.MethodReference
+import org.testeditor.aml.dsl.AmlStandaloneSetup
+import org.testeditor.aml.impl.AmlFactoryImpl
+import org.testeditor.tcl.dsl.tests.parser.AbstractParserTest
+import org.testeditor.tcl.util.TclModelUtil
+
+import static org.mockito.Matchers.*
+
+import static extension org.mockito.Mockito.*
+
+class TclMissingFixtureValidatorTest extends AbstractParserTest {
+
+	@Mock TclModelUtil tclModelUtil // injected into class under test
+	@InjectMocks TclValidator tclValidator // class under test
+	@Mock JvmParameterizedTypeReference typeReferenceMock
+	@Mock ValidationMessageAcceptor messageAcceptor
+
+	var AmlFactory amlFactory
+
+	val message = ArgumentCaptor.forClass(String)
+
+	@Before
+	override void setUp() {
+		super.setUp
+		MockitoAnnotations.initMocks(this)
+
+		val injector = (new AmlStandaloneSetup).createInjectorAndDoEMFRegistration
+		amlFactory = injector.getInstance(AmlFactoryImpl)
+
+		val operationMock = mock(JvmOperation)
+		val jvmTypeMock = mock(JvmType)
+		val interactionTypeMock = mock(InteractionType)
+		val methodReferenceMock = mock(MethodReference)
+
+		when(tclModelUtil.getInteraction(anyObject)).thenReturn(interactionTypeMock)
+		when(interactionTypeMock.defaultMethod).thenReturn(methodReferenceMock)
+		when(methodReferenceMock.operation).thenReturn(operationMock)
+		when(methodReferenceMock.typeReference).thenReturn(typeReferenceMock)
+		when(typeReferenceMock.type).thenReturn(jvmTypeMock)
```

Comment:
When a mock-chain like this is necessary you have two options:
- Refactor the code so that it's easier to mock
- Use mock(class, Answers.RETURNS_DEEP_STUBS) 

If the second approach is necessary you can then write something like `when(interactionTypeMock.defaultMethod.operation.typeReference.type).thenReturn(jvmTypeMock)`

But yeah, this isn't the nicest in the world either ;)